charts/consul/templates/telemetry-collector-deployment.yaml

```diff
@@ -0,0 +1,362 @@
+      {{- toYaml .Values.global.extraLabels | nindent 4 }}
+    {{- end }}
+spec:
+  replicas: {{ .Values.telemetryCollector.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "consul.name" . }}
+      chart: {{ template "consul.chart" . }}
+      release: {{ .Release.Name }}
+      component: consul-telemetry-collector
+  template:
+    metadata:
+      annotations:
+        "consul.hashicorp.com/connect-inject": "false"
+        # This annotation tells the endpoints controller that this pod was injected even though it wasn't. The
+        # endpoints controller would then sync the endpoint into Consul
+        "consul.hashicorp.com/connect-inject-status": "injected"
+        # We aren't using tproxy and we don't have an original pod. This would be simpler if we made a path similar
+        # to gateways
+        "consul.hashicorp.com/transparent-proxy": "false"
+        "consul.hashicorp.com/transparent-proxy-overwrite-probes": "false"
+        "consul.hashicorp.com/connect-k8s-version": {{ $.Chart.Version }}
+        {{- if .Values.telemetryCollector.customExporterConfig }}
+        # configmap checksum
+        "consul.hashicorp.com/config-checksum": {{ include (print $.Template.BasePath "/telemetry-collector-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        # vault annotations
+        {{- if (and .Values.global.secretsBackend.vault.enabled .Values.global.tls.enabled) }}
+        "vault.hashicorp.com/agent-init-first": "true"
+        "vault.hashicorp.com/agent-inject": "true"
+        "vault.hashicorp.com/role": {{ .Values.global.secretsBackend.vault.consulCARole }}
+        "vault.hashicorp.com/agent-inject-secret-serverca.crt": {{ .Values.global.tls.caCert.secretName }}
+        "vault.hashicorp.com/agent-inject-template-serverca.crt": {{ template "consul.serverTLSCATemplate" . }}
+        {{- if and .Values.global.secretsBackend.vault.ca.secretName .Values.global.secretsBackend.vault.ca.secretKey }}
+        "vault.hashicorp.com/agent-extra-secret": "{{ .Values.global.secretsBackend.vault.ca.secretName }}"
+        "vault.hashicorp.com/ca-cert": "/vault/custom/{{ .Values.global.secretsBackend.vault.ca.secretKey }}"
+        {{- end }}
+        {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
+        {{ tpl .Values.global.secretsBackend.vault.agentAnnotations . | nindent 8 | trim }}
+        {{- end }}
+        {{- end }}
+
+      labels:
+        consul.hashicorp.com/connect-inject-managed-by: consul-k8s-endpoints-controller
+        app: {{ template "consul.name" . }}
+        chart: {{ template "consul.chart" . }}
+        release: {{ .Release.Name }}
+        component: consul-telemetry-collector
+        {{- if .Values.global.extraLabels }}
+          {{- toYaml .Values.global.extraLabels | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "consul.fullname" . }}-telemetry-collector
+      initContainers:
+      - name: consul-connect-inject-init
```

Comment:
I think I am missing something. Why are you doing this injection and not letting the webhook do it for you?

Reply:
@curtbushko This is because we don't want to create a dependency between this deployment and the webhook. When intalled in the same release, we cannot guarantee that the webhook will be running to the injection for this deployment. That's why we need to do manual injection

Reply:
Sounds good! Thanks @ishustava